Update ch1-object-basics.md (#7249)
MystenLabs/sui: doc/src/build/programming-with-objects/ch1-object-basics.md

}
```
The above `struct` defines a data structure that can represent RGB color. `struct`s like this can be used to organize data with complicated semantics. However, instances of `struct`s like `Color` are not Sui objects yet.

> To define a struct that represents a Sui object type, we must add a `key` capability to the definition, and the first field of the struct must be the `id` of the object with type `UID` from the [object module](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/sources/object.move). A module from the core [Sui Move package/framework](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/Move.toml).

Reviewer: ```suggestion
To define a struct that represents a Sui object type, we must add a `key` capability to the definition, and the first field of the struct must be the `id` of the object with type `UID` from the [object module](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/sources/object.move) - a module from the core [Sui Framework](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/Move.toml).
```

Now `ColorObject` represents a Sui object type and can be used to create Sui objects that can be eventually stored on the Sui chain.
> :books: In both core Move and Sui Move, the [key ability](https://github.com/move-language/move/blob/main/language/documentation/book/src/abilities.md#key) denotes a type that can appear as a key in global storage. However, the structure of global storage is a bit different: core Move uses a (type, `address`)-indexed map, whereas Sui Move uses a map keyed by object IDs.

> > :bulb: The `UID` type is internal to Sui, and you most likely won't need to deal with it directly. For curious readers, it contains the "unique ID" that defines an object on the Sui chain. It is unique in the sense that no two values of type `UID` will ever have the same underlying set of bytes.

Reviewer: ```suggestion
> :bulb: The `UID` type is internal to Sui, and you most likely won't need to deal with it directly. For curious readers, it contains the "unique ID" that defines an object on the Sui network. It is unique in the sense that no two values of type `UID` will ever have the same underlying set of bytes.
```

```rust
public fun transfer<T: key>(obj: T, recipient: address)
```

> This places `obj` in global storage along with metadata that records `recipient` as the owner of the object. In Sui, every object must have an owner, which can be either an address, another object, or "shared"--see [object ownership](https://github.com/MystenLabs/sui/blob/main/doc/src/build/move/sui-move-library.md) for more details.

Reviewer: ```suggestion
This places `obj` in global storage along with metadata that records `recipient` as the owner of the object. In Sui, every object must have an owner, which can be either an address, another object, or "shared"--see [Object ownership](../learn/objects#object-ownership) for more details.
```
